src/core_tests/replay_flag.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+    machines::WorkflowMachines,
+    protos::coresdk::workflow_commands::StartTimer,
+    protos::temporal::api::enums::v1::CommandType,
+    test_help::canned_histories,
+    test_workflow_driver::{CommandSender, TestWorkflowDriver},
+    workflow::WorkflowManager,
+};
+use rstest::{fixture, rstest};
+use std::time::Duration;
+
+fn timers_wf(num_timers: usize) -> TestWorkflowDriver {
+    TestWorkflowDriver::new(|mut command_sink: CommandSender| async move {
+        for tnum in 1..=num_timers {
+            let timer = StartTimer {
+                timer_id: format!("timer-{}", tnum),
+                start_to_fire_timeout: Some(Duration::from_secs(1).into()),
+            };
+            command_sink.timer(timer).await;
+        }
+        command_sink.complete_workflow_execution();
+    })
+}
+
+#[fixture(num_timers = 1)]
+fn fire_happy_hist(num_timers: usize) -> WorkflowMachines {
+    let twd = timers_wf(num_timers);
+    // Add 1 b/c history takes # wf tasks, not timers
+    let t = canned_histories::long_sequential_timers(num_timers + 1);
+    WorkflowMachines::new(
+        "wfid".to_string(),
+        "runid".to_string(),
+        t.as_history_update(),
+        Box::new(twd).into(),
+    )
+}
+
+#[rstest]
+#[case::one_timer(fire_happy_hist(1), 1)]
+#[case::five_timers(fire_happy_hist(5), 5)]
+fn replay_flag_is_correct(#[case] setup: WorkflowMachines, #[case] num_timers: usize) {
```

Comment:
nit. `is_correct` is slightly misleading should we follow given-when-then naming for tests, or at least when-then?

Reply:
Not a huge fan of those naming schemes, because this one would be something like `given_a_history_with_multiple_workflow_tasks_when_polling_and_completing_then_replay_is_true_until_the_last_wft_started`

Which makes test result output pretty awful sometimes. But I'm happy to just add that as a comment at the top of the test. If we want to do a policy of that I'd be happy to. I know I'd like to see them in @bergundy 's node tests and he'd probably like to see them in my tests :)